src/lib/db.ts

```diff
@@ -83,29 +108,57 @@ export class DB {
 
   // retrieves data from `shared-tables/${shareId}/items`
   async getAllItems() {
-    const itemsRef = this.shareRef && this.shareRef.child("items");
+    const itemsRef = this.getItemsRef();
     return itemsRef && itemsRef.once("value");
   }
 
   async getUserItems(userLabel: string) {
-    const itemsRef = this.shareRef && this.shareRef.child("items");
-    const userItemsRef = itemsRef && itemsRef.child(userLabel);
+    const userItemsRef = this.getItemsRefForUser(userLabel);
     return userItemsRef && userItemsRef.once("value");
   }
 
   setUserItems(userLabel: string, items: any) {
-    const itemsRef = this.shareRef && this.shareRef.child("items");
-    const userItemsRef = itemsRef && itemsRef.child(userLabel);
+    const userItemsRef = this.getItemsRefForUser(userLabel);
     if (userItemsRef && items) {
       userItemsRef.set(items);
     }
   }
 
   async getConnectedUsers() {
-    const connectedRef = this.shareRef && this.shareRef.child("connectedUsers");
+    const connectedRef = this.getConnectedUsersRef();
     return connectedRef && connectedRef.once("value");
   }
 
+  installUserItemListeners() {
+    const connectedRef = this.getConnectedUsersRef();
+    connectedRef && connectedRef.on("child_added", userData => {
```

Comment:
I think this needs a `.once("value",` in order to add listeners for all the existing users.